server/consumer.go

```diff
@@ -2359,6 +2354,15 @@ func (o *consumer) deliverMsg(dsubj, subj string, hdr, msg []byte, seq, dc uint6
 
 	// FIXME(dlc) - Capture errors?
 	o.updateDelivered(dseq, seq, dc, ts)
+
+	// If we are ack none and mset is interest only we should make sure stream removes interest.
+	if ap == AckNone && mset.cfg.Retention != LimitsPolicy {
+		if o.node == nil || o.cfg.Direct {
+			mset.amch <- seq
```

Comment:
You used to check for `amch != nil`, not sure if this was necessary or not, but you don't do that anymore.

Reply:
In stream.go we always create when cfg.Retention != LimitsPolicy, and since we capture that condition above in line 2359 it has to be present.

Reply:
As long as it is not set to nil (which I just checked it is not), then indeed it is ok.